torchtitan/train.py

```diff
@@ -43,6 +45,11 @@ def main(job_config: JobConfig):
     # take control of garbage collection to avoid stragglers
     gc_handler = utils.GarbageCollection(gc_freq=job_config.training.gc_freq)
 
+    device_module, device_type = utils.device_module, utils.device_type
+    device = torch.device(f"{device_type}:{int(os.environ['LOCAL_RANK'])}")
+    device_module.set_device(device)
+    ft_manager = init_ft_manager(job_config)
```

Comment:
curious, whats the reason for moving the `device_module.set_device`. Does `init_ft_manager` depend on this?

Reply:
ye, that's the reason.  I just added comments.